slippy/tile_grid.go

```diff
@@ -0,0 +1,107 @@
+package slippy
+
+import (
+	"fmt"
+	"math"
+)
+
+// TileGrid contains the tile layout, including ability to get WGS84 coordinates for tile extents
+type TileGrid interface {
```

Comment:
I believe that we'd still need a mechanism to support getting a tile's extent in both projected as well as WGS84 units to support downstream usage within Tegola.  Part of the reasoning for separating tiling logic out to support equirectangular tile schemes is in support of tile rendering as "4326" and a non-square tile grid.  But, again, I very well could be missing some of the vision for these changes.  I'd love to chat further so that I'm on the same page!

Reply:
I was thinking that the conversion from one coordinate system to another should be decoupled from `slippy` (it was not before, but it is something we were looking to change). The implementer of `Grid` works only with its native units and then the calling code can use the `proj` package to convert to WGS84 or other projection systems.

Reply:
Gotcha. Makes sense.

Reply:
@ear7h re: `you can compare tiles directly`, I didn't realize that structs in golang offered an ordering mechanism (or operator overloading) but am having trouble a mechanism to support this construct.  Is there something I'm missing or misunderstanding?

Reply:
@meilinger My mistake, that doesn't work 😬 They can be compared for equality but not ordering.